apps/dotcom/src/utils/url.ts

```diff
@@ -1,3 +1,26 @@
 export function openUrl(url: string) {
 	window.open(url, '_blank')
 }
+
+/**
+ * Compares two URLs to see if they are the same. This ignores encoding.
+ * @param first - The first URL to compare.
+ * @param second - The second URL to compare.
+ * @returns Whether the URLs are the same.
+ */
+export function sameUrls(first: string, second: string): boolean {
```

Comment:
ach, I'm gonna be pedantic on this a bit :)

we shouldn't do this, because `decodeURIComponent` is meant to operate only on individual part of a search param.

so, this:
`decodeURIComponent(`?foo=a%3D1%26b%3D2%26c%3D3`)`
shouldn't actually be equate to:
`decodeURIComponent(`?foo=a=1&b=2&c=3`)`
because they're actually two different things.

The first is saying `foo` is equal to `a=1&b=2&c=3` (encoded)
and the second is saying `foo` = `a` and `b` = `2` and `c` = `3` (and yeah it's prbly a bit of a parser error there but it's a contrived example)

This is because you can have encoded things like this in URLs (security stuff aside here for a sec):
`https://example.com/?redirectTo=https%3A%2F%2Ffoo.com%3Fbaz%3Dbar`

Anyway, we shouldn't do this! It'll have subtle footguns in there!

Reply:
It's all cursed 😂 Removed it.

Reply:
hehe agreed. thx :D